src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -867,19 +867,11 @@ internal void ClearMarkers(ScriptFile scriptFile)
                 fileLock.Release();
             }
 
-
-            var uriBuilder = new UriBuilder
-            {
-                Scheme = Uri.UriSchemeFile,
-                Path = scriptFile.FilePath,
-                Host = string.Empty,
-            };
-
             // Always send syntax and semantic errors.  We want to
             // make sure no out-of-date markers are being displayed.
             _languageServer.Document.PublishDiagnostics(new PublishDiagnosticsParams()
             {
-                Uri = uriBuilder.Uri,
+                Uri = new Uri(scriptFile.DocumentUri),
```

Comment:
Theoretically `UriBuilder` exists because `Uri` is supposed to be immutable (meaning you can safely expose a reference to an internally used one). So you could just return `scriptFile.DocumentUri` here. But there might be some merit in cloning it instead.

Reply:
Well `PublishDiagnosticsParams.Uri` expects a `Uri` type and `scriptFile.DocumentUri` is a string.

Reply:
we talked offline about this. Rob was under the impression that `DocumentUri` was a `Uri` but it's a `string`... it should really be a `Uri`...